packages/fuels-macros/src/setup_program_test/parsing/validations.rs

```diff
@@ -0,0 +1,112 @@
+use std::collections::HashSet;
+
+use fuels_code_gen::ProgramType;
+use proc_macro2::Span;
+use syn::{Error, LitStr, Result};
+
+use crate::{
+    parse_utils::ErrorsExt,
+    setup_program_test::parsing::{
+        AbigenCommand, DeployContractCommand, InitializeWalletCommand, LoadScriptCommand,
+    },
+};
+
+pub(crate) fn extract_the_abigen_command(
+    parent_span: Span,
+    abigen_commands: &[AbigenCommand],
+) -> Result<AbigenCommand> {
+    match abigen_commands {
+        [single_command] => Ok(single_command.clone()),
+        commands => {
+            let err = commands
+                .iter()
+                .map(|command| Error::new(command.span, "Only one `Abigen` command allowed"))
+                .combine_errors()
+                .unwrap_or_else(|| Error::new(parent_span, "Add an `Abigen(..)` command!"));
+
+            Err(err)
+        }
+    }
+}
+
+pub(crate) fn validate_all_contracts_are_known(
+    abigen_command: &AbigenCommand,
+    deploy_contracts: &[DeployContractCommand],
+) -> Result<()> {
+    contracts_to_deploy(deploy_contracts)
```

Comment:
IMO we should leave it as is. We do not gain  any clarity from in-lining it and we would have to specify that we are collecting to `Hashset<&LitStr>`.

Reply:
Ok I understand the rationale, I guess I'm still getting used to shallow methods haha

Reply:
Having this shallow method, in this case, is nice for readability. However, I would suggest slightly renaming it; `contracts_to_deploy` doesn't tell me right away that it is extracting something out of `deploy_contracts`, so I have to go to its definition to see that it's _extracting_ something. It could be even more readable and descriptive of what it does if it were written as `extract_contracts_to_deploy(deploy_contracts)`.

This is a minor readability nit though, so take it with a grain of salt ❤️